staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -275,7 +301,320 @@ func TestDecode(t *testing.T) {
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e.CBORType != "primitives" || e.GoType != "string" {
+					t.Errorf("Expected types: primitives,string. Got: %#v,%#v", e.CBORType, e.GoType)
+				}
+			}),
+		},
+		{
+			name:          "map with byte string key type",
+			in:            hex("a143abcdef187b"), // {h'ABCDEF': 123}
+			want:          map[string]interface{}{"\xab\xcd\xef": int64(123)},
+			assertOnError: assertNilError,
+		},
+		{
+			name:  "max positive integer value supported by cbor: 2^64 - 1",
+			modes: []cbor.DecMode{modes.Decode},
+			in:    hex("1bffffffffffffffff"), // 18446744073709551615
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e == nil {
+					t.Error("Expected non-nil error. ")
+				} else if !strings.Contains(e.Error(), "cbor: cannot unmarshal positive integer into Go value of type int64 (18446744073709551615 overflows Go's int64)") {
+					t.Errorf("Expected overflow error. Got: %s", e.Error())
+				}
+			}),
+		},
+		{
+			name: "min negative integer value supported by cbor: -2^64",
+			in:   hex("3bffffffffffffffff"), // -18446744073709551616
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.UnmarshalTypeError) {
+				if e == nil {
+					t.Error("Expected non-nil error. ")
+				} else if !strings.Contains(e.Error(), "cbor: cannot unmarshal negative integer into Go value of type int64 (-18446744073709551616 overflows Go's int64)") {
+					t.Errorf("Expected overflow error. Got: %s", e.Error())
+				}
+			}),
+		},
+		{
+			name:          "tag 0 timestamp string",
+			in:            hex("c074323030362d30312d30325431353a30343a30355a"), // 0("2006-01-02T15:04:05Z")
+			want:          "2006-01-02T15:04:05Z",
+			failureReason: "decoding cbor data tagged with 0 produces time.Time instead of RFC3339 timestamp string",
+			assertOnError: assertNilError,
+		},
+		{
+			name: "tag 0 with byte string",
+			in:   hex("c043414243"), // 0('ABC')
+			assertOnError: func(t *testing.T, e error) {
+				if e == nil {
+					t.Error("Expected non-nil error. ")
+				} else if !strings.Contains(e.Error(), "cbor: tag number 0 must be followed by text string, got byte string") {
+					t.Errorf("Expected error: %s, got: %s \n", "cbor: tag number 0 must be followed by text string, got byte string", e.Error())
+				}
+			},
+			want: UncheckedValue{},
+		},
+		{
+			name: "tag 0 with invalid text string",
+			in:   hex("c06474657874"), // 0("text")
+			assertOnError: func(t *testing.T, e error) {
+				if e == nil {
+					t.Error("Expected non-nil error. ")
+				} else if !strings.Contains(e.Error(), "cbor: cannot set text for time.Time: parsing time") {
+					t.Errorf("Expected error: %s, got: %s \n", "cbor: cannot set text for time.Time: parsing time", e.Error())
+				}
+			},
+			want: UncheckedValue{},
+		},
+		{
+			name:          "tag 1 timestamp unsigned integer",
```

Comment:
what is tag 0 versus tag 1

Reply:
tag 0 contains a time value as a text string (https://www.rfc-editor.org/rfc/rfc8949.html#section-3.4.1)
tag 1 contains a time value as a epoch timestamp (https://www.rfc-editor.org/rfc/rfc8949.html#section-3.4.2)